### PR TITLE
Update basic access control task with new configs

### PR DESCRIPTION
**Pull request by mandarjog**
This PR updates basic access control tasks with new configs.

At present the new adapter and template links are broken because those artifacts have not been generated yet.

### CODE REVIEW

File: _docs/tasks/basic-access-control.md [hunk 1]
```diff
@@ -45,31 +45,56 @@ of the `reviews` service. We would like to cut off access to version `v3` of the
    If you log in as any other user (or logout) you should see red ratings stars with each review,
    indicating that the `ratings` service is being called by the "v3" version of the `reviews` service.
 
-1. Explicitly deny access to version `v3` of the `reviews` service. 
+1. Explicitly deny access to version `v3` of the `reviews` service.
 
-   ```bash
-   istioctl mixer rule create global ratings.default.svc.cluster.local -f samples/apps/bookinfo/mixer-rule-ratings-denial.yaml
+   Before setting up the deny rule, we must create a handler and an instance definition that can be used in the deny rule.
+   Replace `metadata.namespace` with the appropriate value.
```
Comment (kyessenov) [inline _docs/tasks/basic-access-control.md]: Maybe skip and make our istioctl utilize `-n` flag to inject `metadata.namespace`? I think that's what we do for pilot rules.

File: _docs/tasks/basic-access-control.md [hunk 2]
```diff
@@ -45,31 +45,56 @@ of the `reviews` service. We would like to cut off access to version `v3` of the
...
+   ```yaml
+   apiVersion: "config.istio.io/v1alpha2"
+   kind: denier
+   metadata:
+     name: handler
+     namespace: default
+   spec:
+     status:
+       code: 7
```
Comment (kyessenov) [inline _docs/tasks/basic-access-control.md]: Add a comment that it is gRPC status error code.

File: _docs/tasks/basic-access-control.md [hunk 3]
```diff
@@ -45,31 +45,56 @@ of the `reviews` service. We would like to cut off access to version `v3` of the
...
    indicating that the `ratings` service is being called by the "v3" version of the `reviews` service.
 
-1. Explicitly deny access to version `v3` of the `reviews` service. 
+1. Explicitly deny access to version `v3` of the `reviews` service.
 
-   ```bash
-   istioctl mixer rule create global ratings.default.svc.cluster.local -f samples/apps/bookinfo/mixer-rule-ratings-denial.yaml
+   Before setting up the deny rule, we must create a handler and an instance definition that can be used in the deny rule.
+   Replace `metadata.namespace` with the appropriate value.
+   ```yaml
```
Comment (kyessenov) [inline _docs/tasks/basic-access-control.md]: We used to be able to import artifacts from the repo directly. @geeknoid is out, do you remember any examples?

File: _docs/tasks/basic-access-control.md [hunk 4]
```diff
@@ -45,31 +45,56 @@ of the `reviews` service. We would like to cut off access to version `v3` of the
...
+   apiVersion: "config.istio.io/v1alpha2"
+   kind: rule
+   metadata:
+     name: denyreviewsv3
+     namespace: default
+   spec:
+     match: destination.labels["app"] == "ratings" && source.labels["app"]=="reviews" && source.labels["version"] == "v3"
+     actions:
+     - handler: denyall.denier
+       instances: [ denyrequest.checknothing ]
```
Comment (kyessenov) [inline _docs/tasks/basic-access-control.md]: is the value an array or a string that contains an array?

File: _docs/tasks/basic-access-control.md [hunk 5]
```diff
@@ -45,31 +45,56 @@ of the `reviews` service. We would like to cut off access to version `v3` of the
...
 
-1. Explicitly deny access to version `v3` of the `reviews` service. 
+1. Explicitly deny access to version `v3` of the `reviews` service.
 
-   ```bash
-   istioctl mixer rule create global ratings.default.svc.cluster.local -f samples/apps/bookinfo/mixer-rule-ratings-denial.yaml
+   Before setting up the deny rule, we must create a handler and an instance definition that can be used in the deny rule.
+   Replace `metadata.namespace` with the appropriate value.
+   ```yaml
+   apiVersion: "config.istio.io/v1alpha2"
```
Comment (kyessenov) [inline _docs/tasks/basic-access-control.md]: Quotes are unnecessary here: https://github.com/istio/istio/blob/master/samples/apps/bookinfo/rules/route-rule-reviews-50-v3.yaml

File: _docs/tasks/basic-access-control.md (hunk 4 above)
Comment (mandarjog) [reply _docs/tasks/basic-access-control.md]: `instances` is an array.

File: _docs/tasks/basic-access-control.md (hunk 3 above)
Comment (mandarjog) [reply _docs/tasks/basic-access-control.md]: I have changed it to use direct links, here is how you do it.

```
  <pre data-src="https://raw.githubusercontent.com/istio/istio/release-0.1/samples/apps/bookinfo/rules/mixer-rule-ratings-ratelimit.yaml"></pre>
```

File: _docs/tasks/basic-access-control.md (hunk 1 above)
Comment (mandarjog) [reply _docs/tasks/basic-access-control.md]: `istioctl` does not support this yet.

File: _docs/tasks/basic-access-control.md (hunk 4 above)
Comment (douglas-reid) [reply _docs/tasks/basic-access-control.md]: This should match the style of the configs in istio/mixer.

File: _docs/tasks/basic-access-control.md [hunk 6]
```diff
@@ -81,45 +106,57 @@ of the `reviews` service. We would like to cut off access to version `v3` of the
...
+   apiVersion: config.istio.io/v1alpha2
+   kind: rule
+   metadata:
+     name: checkversion
+     namespace: istio-config-default
+   spec:
+     match: destination.labels["app"] == "ratings"
+     actions:
+     - handler: staticversion.listchecker
+       instances: [ appversion.listentry ]
```
Comment (douglas-reid) [inline _docs/tasks/basic-access-control.md]: again, we should strive for consistency on instance listings.

**Comment by ldemailly** [comment]
now it's green. just need someone to accept it

**Comment by mandarjog** [comment]
@douglas-reid updated config examples.

**Comment by mandarjog** [comment]
@kyessenov I updated the PR to use `istioctl` exclusively. Added a link for grpc error codes.

**Comment by kyessenov** [comment]
Awesome, thanks!

**Pull request merged**